### ISSUE TITLE
DM2550ZB Change phase 

### ISSUE THREAD

**Issue by nickb24** [claudegel/sinope-zha#14]
The DM2550ZB is supposed to have the ability to change the phase mode. According to the manual "The DM2550ZB dimmer is configured in reverse phase by default to ensure more compatibility with most LED, incandescent and halogen bulbs. If a flickering problem persists, it is possible to change the control mode in the dimmer settings on the Neviweb app."

What is the best way to help discover this missing attribute?

**Comment by claudegel** [comment]
Please install zha-toolkit via HACS. then use service zha_toolkit.scan_device in devtool/services
```
service: zha_toolkit.scan_device
data:
  ieee: 00:0d:6f:00:05:7d:2d:34 ## your dimmer ieee
  endpoint: 1
  event_success: my_read_success_trigger_event
  event_fail: my_read_fail_trigger_event
  event_done: my_read_done_trigger_event
```
This will scan all cluster of the device and create a file in config/scans/ directory. 
Send the file to me so I can check which attribute could be the one we need.
you can copy paste the file here

**Comment by claudegel** [comment]
If we didn't get the attribute then install my sinope-zha qwirck by following this documentation:
https://github.com/claudegel/sinope-zha
Once installed let me know I'll send you an updated light.py file which contain a list of unknown attributes for cluster 0xff01

**Comment by claudegel** [comment]
Thanks, can you try to change value of cluster 0xff01, attribute 0x0001
The value now is 1, try to change to 0 or 2

**Comment by claudegel** [comment]
Can you send me your zigbee signature for your DM2550ZB
You'll find it in parameters/devices and services/Zigbee Home Automation
select your device and click the three vertical dot near reconfigure
tehn choose manage zigbee device and signature.
You are missing cluster 0x0008. It is possible that your device is not properly detected by ZHA

**Comment by claudegel** [comment]
You can't imagine how many time I've busted all my HA setup during development of my custom components :)

**Comment by claudegel** [comment]
I just realize that the scan you've sent me was for a SW2500ZB not your DM2550ZB :(
In z2m the cluster 0xff01, attribute 0x0001, 0x0010, 0x0012 are not available for your dimmer and light

**Comment by claudegel** [comment]
I'm not familiar with z2m but what is the Clusters listed on top menu of the picture you just sent. Does it list all cluster of your device ?

**Comment by claudegel** [comment]
z2m is based on zigbee-herdsman cluster library and the attributes 0x0001 is not defined in that library. That,s why I prefer ZHA where I can develop my own library and test faster.
maybe there is a way to add those attributes in your herdsman library locally. I'll check if I can find a way to edit that library locally.

**Comment by claudegel** [comment]
unknown in your case is attribute 0x0075 and it is not supported for lights and dimmer

**Comment by claudegel** [comment]
Ok I can see that your dimmer is correctly detected. But don't know why but in the scan, the cluster 0xff01, manufacturer cluster, there is no attributes reported. Are you using my quirks or just ZHA ?
Your SW2500ZB scan was reporting attributes for the cluster 0xff01. Can you tell me if your DM2550ZB was previously connected to Neviweb. Same for your SW2500ZB ?
I need to investigate why there is no attribute for your DM2550ZB. Maybe the manufacturer cluster attributes are populated the first time the device is connected to Neviweb as they are performing firmware update and we do not have access to firmware update in HA.
I know that there was a bug in zha-toolkit that was not reading the manufacturer cluster attribute during the scan. Are you able to see attribute value in zha with the manage zigbee device ?

**Comment by claudegel** [comment]
Ok to get the test attribute you will need to install my sinope-zha.
just create a dir in config/ named zhaquirks and copy inside the file light.py that you will find in the bellow branch
https://github.com/claudegel/sinope-zha/tree/dimmer-test
In your configuration.yaml add this
```
zha_toolkit:

zha:
  database_path: zigbee.db
  custom_quirks_path: /config/zhaquirks/
```
probably you already have the line zha_toolkit: and zha:
the important line is custom_quirks...
restart HA
this will allow to load light.py instead of the one in zha and you will be able to play with the attributes for what we are looking for

**Comment by claudegel** [comment]
For the SW2500ZB that you have scanned at the beginning did that devive was connected to neviweb before ?
In the manage ZigBee device section you see the attribute but can you get value for the attributes ?

**Comment by claudegel** [comment]
When you put 0 or 2 for attribute 0x0001 when you read it back does the value change.
This attribute is of type bool so I think you should just put false
This way you will get Bool.false

So if you never setup your SW2500ZB in Neviweb than zha-toolkit scan is the problem for your DM2550ZB as it can't scan the attributes for 0xff01

Do you see attribute 0x0010, 0x0012 and 0x0013 ? especially the last two as they are of type enum   and it is possible to put 0, 1 or 2

**Comment by claudegel** [comment]
Yes, if when you read value it say None it is because this attribute is not supported by your device. If youget something esle then it is a supported attribute.
Just to let you know, tomorrow I,ll be out until friday night. so it is possible that I can't answer. But I'll be back friday night.

**Comment by claudegel** [comment]
I'm thinking of buying one so I can put it in Neviweb and sniff the zigbee communications. That's how I found most attributes

**Comment by claudegel** [comment]
I just bought a DM2550ZB so I'll be able to sniff all the zigbee trafic between Neviweb and the device.
Also it is possible that the attribute we are looking for is not in the 0xff01 cluster. Sinopé could have add it to another cluster like lighting, metering or level control. In that last one there is an attribute 0x000f named options. It could be this.

**Comment by claudegel** [comment]
If I can find this it would be great. I'm gonna sniff to follow all the connection process.

**Comment by claudegel** [comment]
Finally I got the attribute for phase control, 0xff01 0x0056
I also found attribute for key double up action. When set to on double up key press set light intensity to 100%
0xff01 0x0058
I've updated branche dimmer-test, file light.py
Please test as my DM2550ZB is on Neviweb now to update my code for neviweb130
let me know

**Comment by claudegel** [comment]
if you already download the new light.py I just found a typo. Please download it again. Sorry

**Comment by claudegel** [comment]
Did you had time to test the double clic up. When activated via cluster 0xff01, attribute 0x0058. With the double clic up the light turn on at 100% intensity. I also found that attribute 0x0090 is a kind of power watt summation and could be used for consumption statistic. I need to test a little bit more but I'm close to make a PR. There is also attribute 0x0010 where the device store its last on intensity. Could you change the intensity and check if there are the same change in attribute 0x0010 value ?
Thank you for your help.

**Comment by claudegel** [comment]
I'm presently finishing digging for the DM2550ZB and Sedna valve with flow meter attributes and I'll send a PR to ZHA and z2m to add those attributes.
Thank you for helping me testing all that.

**Comment by claudegel** [comment]
I just finish sedna flow meter integration and will push a PR soon in ZHA
